File: BSvZP-Common/Messages/ReadyReply.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Common;

namespace Messages
{
    public class ReadyReply : Reply
    {
        #region Private Properties
        private static Int16 ClassId { get { return (Int16)MESSAGE_CLASS_IDS.ReadyReply; } }
        #endregion

        #region Public Properties
        public override Message.MESSAGE_CLASS_IDS MessageTypeId() { return (Message.MESSAGE_CLASS_IDS)ClassId; }

        public static new int MinimumEncodingLength
        {
            get
            {
                return 4;                // Object header
            }
        }
        #endregion
        
        #region Constructors and Factory Methods
        /// <summary>
        /// Default message constructor, used by Factory methods (i.e., the Create methods)
        /// </summary>

        protected ReadyReply() { }

        /// <summary>
        /// Constructor used by all specializations, which in turn are used by
        /// senders of a message 
        /// </summary>
        /// <param name="conversationId">conversation id</param>
        /// <param name="status">Status of the ack/nak</status>
        /// <param name="note">error message or note</note>
        public ReadyReply(PossibleStatus status, string note = null) :
            base(Reply.PossibleTypes.ReadyReply, status, note)
        {
        }

        /// <summary>
        /// Factor method to create a message from a byte list
        /// </summary>
        /// <param name="messageBytes">A byte list from which the message will be decoded</param>
        /// <returns>A new message of the right specialization</returns>
        new public static ReadyReply Create(ByteList messageBytes)
        {
            ReadyReply result = null;

            if (messageBytes==null || messageBytes.RemainingToRead<MinimumEncodingLength)
                throw new ApplicationException("Invalid message byte array");
            if (messageBytes.PeekInt16() != ClassId)
                throw new ApplicationException("Invalid message class id");
            else
            {
                result = new ReadyReply();
                result.Decode(messageBytes);
            }

            return result;
        }
        #endregion

        #region Encoding and Decoding methods

        override public void Encode(ByteList bytes)
        {
            bytes.Add(ClassId);                           // Write out this class id first

            Int16 lengthPos = bytes.CurrentWritePosition;   // Get the current write position, so we
                                                                // can write the length here later
            bytes.Add((Int16) 0);                           // Write out a place holder for the length

            base.Encode(bytes);                             // Encode stuff from base class

            Int16 length = Convert.ToInt16(bytes.CurrentWritePosition - lengthPos - 2);
            bytes.WriteInt16To(lengthPos, length);          // Write out the length of this object        

        }

        override public void Decode(ByteList bytes)
        {
            Int16 objType = bytes.GetInt16();
            Int16 objLength = bytes.GetInt16();

            bytes.SetNewReadLimit(objLength);

            base.Decode(bytes);

            bytes.RestorePreviosReadLimit();
        }

        #endregion

    }
}

File: BSvsZP-Common -Java/Messages/ReadyReply.java
package Messages;

import java.io.NotActiveException;
import java.net.UnknownHostException;

import org.omg.CORBA.portable.ApplicationException;

import Common.ByteList;

public class ReadyReply extends Reply
{
	private static short ClassId; // { get { return  } }
	private static int MinimumEncodingLength;
  
	
        @Override
	public short getClassId() {
		ClassId = (short)MESSAGE_CLASS_IDS.ReadyReply.getValue();
		return   ClassId;
	}

	protected ReadyReply() { }
	
	public ReadyReply(PossibleStatus status, String...note)
	{
		super(Reply.PossibleTypes.ReadyReply, status, (note.length ==1 ? note[0] : null));
    }
	 
	
	public static ReadyReply Create(ByteList messageBytes) throws Exception
    {
        ReadyReply result = null;

        if (messageBytes==null || messageBytes.getRemainingToRead()< getMinimumEncodingLength())
            throw new ApplicationException("Invalid message byte array", null);
        if (messageBytes.PeekInt16() != ClassId)
            throw new ApplicationException("Invalid message class id", null);
        else
        {
            result = new ReadyReply();
            result.Decode(messageBytes);
        }

        return result;
    }
	
	 @Override public void Encode(ByteList bytes) throws NotActiveException, UnknownHostException, Exception
     {
         bytes.Add((short)MESSAGE_CLASS_IDS.ReadyReply.getValue());                           // Write out this class id first
         bytes.update();
         short lengthPos = bytes.getCurrentWritePosition();   // Get the current write position, so we
         bytes.update();                                           // can write the length here later
         bytes.Add((short) 0);                           // Write out a place holder for the length
         bytes.update();
         super.Encode(bytes);                             // Encode stuff from base class
         bytes.update();
         short length = (short)(bytes.getCurrentWritePosition() - lengthPos - 2);
         bytes.WriteInt16To(lengthPos, length);          // Write out the length of this object        
         bytes.update();
     }

     @Override public void Decode(ByteList bytes) throws Exception
     {
         short objType = bytes.GetInt16();  bytes.update();
         short objLength = bytes.GetInt16();  bytes.update();

         bytes.SetNewReadLimit(objLength);
         bytes.update();
         super.Decode(bytes);
         bytes.update();
         bytes.RestorePreviosReadLimit(); bytes.update();
     } 
     
	public static int getMinimumEncodingLength() {
		 MinimumEncodingLength = 4;
		 return MinimumEncodingLength;
	}

	@Override
	public int compareTo(Object o) {
		return 0;
	}

	@Override
	public MESSAGE_CLASS_IDS MessageTypeId() {
		return Message.MESSAGE_CLASS_IDS.fromShort(ClassId);
	}

}
